src/renderer/fonts/font_options.rs

```diff
@@ -15,47 +29,44 @@ pub struct FontOptions {
 }
 
 impl FontOptions {
-    pub fn parse(guifont_setting: &str) -> FontOptions {
+    pub fn parse(guifont_setting: &str) -> Result<FontOptions, &str> {
         let mut font_options = FontOptions::default();
 
-        let mut parts = guifont_setting.split(':').filter(|part| !part.is_empty());
+        let mut parts = guifont_setting
+            .split(FONT_OPTS_SEPARATOR)
+            .filter(|part| !part.is_empty());
 
         if let Some(parts) = parts.next() {
-            let parsed_font_list: Vec<String> = parts
-                .split(',')
+            let parsed_font_list = parts
+                .split(FONT_LIST_SEPARATOR)
                 .filter(|fallback| !fallback.is_empty())
                 .map(parse_font_name)
-                .collect();
+                .collect_vec();
 
             if !parsed_font_list.is_empty() {
                 font_options.font_list = parsed_font_list;
             }
         }
 
         for part in parts {
-            if let Some(hinting_string) = part.strip_prefix("#h-") {
-                font_options.hinting = FontHinting::parse(hinting_string);
-            } else if let Some(edging_string) = part.strip_prefix("#e-") {
-                font_options.edging = FontEdging::parse(edging_string);
-            } else if part.starts_with('h') && part.len() > 1 {
-                if part.contains('.') {
-                    font_options.allow_float_size = true;
-                }
-                if let Ok(parsed_size) = part[1..].parse::<f32>() {
-                    font_options.size = points_to_pixels(parsed_size);
-                }
-            } else if part.starts_with('w') && part.len() > 1 {
-                if let Ok(parsed_size) = part[1..].parse::<f32>() {
-                    font_options.width = points_to_pixels(parsed_size);
-                }
-            } else if part == "b" {
+            if let Some(hinting_string) = part.strip_prefix(FONT_HINTING_PREFIX) {
+                font_options.hinting = FontHinting::parse(hinting_string)?;
+            } else if let Some(edging_string) = part.strip_prefix(FONT_EDGING_PREFIX) {
+                font_options.edging = FontEdging::parse(edging_string)?;
+            } else if part.starts_with(FONT_HEIGHT_PREFIX) && part.len() > 1 {
+                font_options.allow_float_size |= part[1..].contains(ALLOW_FLOAT_SIZE_OPT);
+                font_options.size = parse_pixels(part).map_err(|_| INVALID_SIZE_ERR)?;
+            } else if part.starts_with(FONT_WIDTH_PREFIX) && part.len() > 1 {
+                font_options.allow_float_size |= part[1..].contains(ALLOW_FLOAT_SIZE_OPT);
+                font_options.width = parse_pixels(part).map_err(|_| INVALID_WIDTH_ERR)?;
+            } else if part == FONT_BOLD_OPT {
                 font_options.bold = true;
-            } else if part == "i" {
+            } else if part == FONT_ITALIC_OPT {
                 font_options.italic = true;
             }
```

Comment:
It would be good to give an error for an unknown part. But I'm not sure if we can do that, since that might be valid for another UI.

Reply:
mmm, yeah not sure about that, but maybe it is a good decision.  
On one hand, as you say, it may be valid for another UI and invalid for neovide. On the other hand though, perhaps it is the user's responsibility to adjust their `guifont` setting according to the UI being used. Even though it isn't really standard, pretty much every alternative UI I know exposes some way of knowing if their active or not, and such settings can be conditionally tweaked that way.

@Kethku thoughts?

Reply:
I think we can add more validation later if needed.